### PR TITLE
Add troubleshooting section to user guide

### PR DESCRIPTION
**Pull request by lunaryorn**
Implements #1100, still WIP.

### CODE REVIEW

**Comment by lunaryorn** [comment]
Blocked by #1124, because that issue affects screenshots of the verification buffer.

**Comment by lunaryorn** [comment]
So, finished it up.  LGTM, for the record. 

@cpitclaudel @fmdkdd What's your opinion?

File: doc/user/troubleshooting.rst [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+Newer releases of Flycheck may require newer versions of syntax checking tools.
+For instance Flycheck might now pass a command line flag that older versions do
+not understand, or attempt to parse an updated output format.  In these cases
+the syntax checker will show an error message about an unknown flag, or emit
+output that Flycheck does not understand, which prompts Flycheck to warn that
+even though the syntax checker appeared to not have successfully checked the
+buffer content there are no errors to be found.
+
+If you *are* using the version then this message most likely indicates a flaw in
```
Comment (fmdkdd) [inline doc/user/troubleshooting.rst]: 'using the version': missing a 'latest' here?

File: doc/user/troubleshooting.rst [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  this warning.
+
+* *Red* items indicate *bad* configuration.  `python-pylint` wan’t found in the
+  screenshot, so you’ll not be able to use pylint in the current buffer.
+
+Debug syntax checkers
+=====================
+
+If a syntax checker fails although it successfully verified you need to take a
+closer look.  Flycheck provides you with a command that let’s you run a single
```
Comment (fmdkdd) [inline doc/user/troubleshooting.rst]: `let's` -> `lets`

**Comment by fmdkdd** [comment]
@lunaryorn Looks comprehensive enough, and a good place to expand if the need arises.  Maybe we should add a link to the troubleshooting section in the Github issue template as well?

File: doc/user/troubleshooting.rst (hunk 2 above)
Comment (lunaryorn) [inline doc/user/troubleshooting.rst]: Fixed, thanks

**Comment by lunaryorn** [comment]
@fmdkdd Good idea, I added a paragraph to the issue template.

File: doc/index.rst [hunk 3]
```diff
@@ -46,10 +46,25 @@ run ``M-x eval-buffer``:
 
    (global-flycheck-mode)
 
+*On MacOS* also add the following to :ref:`fix your $PATH environment variable
+<flycheck-macos-exec-path-from-shell>`:
+
+.. code-block:: cl
+
+   (package-install 'exec-path-from-shell)
```
Comment (cpitclaudel) [inline doc/index.rst]: Will this call always work? Doesn't it need a package refresh first?

File: doc/user/installation.rst [hunk 4]
```diff
@@ -39,7 +39,14 @@ Many of these programs are available in the package repositories of Linux
 distributions or in Homebrew_ for macOS.  Others can be installed with standard
 package managers such as Rubygems, NPM, Cabal, etc.
 
+.. important::
+
+   For a GUI Emacs on MacOS we recommend to install and configure
+   exec-path-from-shell_ to make Emacs use the proper ``$PATH`` and avoid a
```
Comment (cpitclaudel) [inline doc/user/installation.rst]: Should `exec-path-from-shell` be monospaced?

File: doc/user/troubleshooting.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  :ref:`flycheck-checker-config-files` has more information about configuration
+  files.
+
+  Likewise the buffer warns you that a ``demo`` syntax checker (which is not
+  part of Flycheck of course) isn’t registered in `flycheck-checkers`.  If you’d
+  like Flycheck to automatically use this syntax checker you should fix this
+  issue by adding it to `flycheck-checkers` but otherwise it’s safe to ignore
+  this warning.
+
+* *Red* items indicate *bad* configuration.  `python-pylint` wan’t found in the
```
Comment (cpitclaudel) [inline doc/user/troubleshooting.rst]: typo: wan't → wasn't

**Comment by cpitclaudel** [comment]
LGTM; great job!

File: doc/user/installation.rst (hunk 4 above)
Comment (lunaryorn) [inline doc/user/installation.rst]: I find monospaced links somewhat strange to read, and in any case you can't easily monospace links in Rest.

File: doc/index.rst (hunk 3 above)
Comment (lunaryorn) [inline doc/index.rst]: We're suggesting a `(package-refresh-contents)` in the code above, and it's the quickstart section. IOW, the user is supposed to type all that into scratch, in the same Emacs session, to try Flycheck.

Comment (cpitclaudel) [inline doc/index.rst]: Oh, of course, thanks!

File: doc/user/troubleshooting.rst (hunk 5 above)
Comment (lunaryorn) [inline doc/user/troubleshooting.rst]: Fixed

**Comment by lunaryorn** [comment]
I fixed the typo; will wait for the Travis CI build and then merge :)

**Comment by lunaryorn** [comment]
And in :sunglasses:

**Pull request merged**